### PR TITLE
pkg/trace: remove sublayer metrics computation

### PR DESCRIPTION
**Pull request by piochelepiotr**
We now generate sublayer metrics in the backend.
That way, we can drop that expensive computation from the agent.
This will also allow us to support distributed sublayer metrics.

### CODE REVIEW

File: pkg/trace/agent/agent.go [hunk 1]
```diff
@@ -234,25 +233,11 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
...
-		}
 		sinputs = append(sinputs, stats.Input{
-			Trace:         pt.WeightedTrace,
-			Sublayers:     pt.Sublayers,
-			Env:           pt.Env,
-			SublayersOnly: p.ClientComputedStats,
+			Trace: pt.WeightedTrace,
+			Env:   pt.Env,
 		})
+		// TODO[piochelepiotr] Maybe we can skip some computation if stats are computed in the tracer and the trace is droped.
```
Comment (gbbr) [inline pkg/trace/agent/agent.go:240]:
```suggestion
		// TODO(piochelepiotr): Maybe we can skip some computation if stats are computed in the tracer and the trace is droped.
```

**Review by gbbr — APPROVED** [review]
<3

File: pkg/trace/agent/agent.go [hunk 2]
```diff
@@ -234,25 +233,11 @@ func (a *Agent) Process(p *api.Payload, sublayerCalculator *stats.SublayerCalcul
...
-					stats.SetSublayersOnSpan(subtrace.Root, subtraceSublayers)
-				}
-			}
-		}
 		sinputs = append(sinputs, stats.Input{
-			Trace:         pt.WeightedTrace,
-			Sublayers:     pt.Sublayers,
-			Env:           pt.Env,
-			SublayersOnly: p.ClientComputedStats,
+			Trace: pt.WeightedTrace,
```
Comment (furmmon) [inline pkg/trace/agent/agent.go:237]: if ClientComputedStats is set, you can actually stop filling sinputs and not allocate it

**Review by raphaelgavache — APPROVED** [review]

File: releasenotes/notes/apm-remove-sublayers-92f2cfdd0edd872b.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    APM: Sublayer metrics (trace.<SPAN_NAME>.duration and derivatives) computation
+    is removed from the agent in favor of new sublayer metrics generated in the backend.
+enhancements:
+  - |
+    APM: Removing the computation of sublayer metrics is greatly improving CPU and memory usage of the trace agent.
```
Comment (omris1) [inline releasenotes/notes/apm-remove-sublayers-92f2cfdd0edd872b.yaml:15]: > Removal of sublayer metric computation improves performance overhead in the trace agent (CPU and memory).

File: releasenotes/notes/apm-remove-sublayers-92f2cfdd0edd872b.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+deprecations:
+  - |
+    APM: Sublayer metrics (trace.<SPAN_NAME>.duration and derivatives) computation
+    is removed from the agent in favor of new sublayer metrics generated in the backend.
```
Comment (omris1) [inline releasenotes/notes/apm-remove-sublayers-92f2cfdd0edd872b.yaml:12]: Need to link to a public doc with a description of the new metric - preferably an addition here: https://docs.datadoghq.com/tracing/guide/metrics_namespace/

**Pull request merged**